php/php.editor/tools/ASTPHP5Scanner.flex

```diff
@@ -943,6 +944,12 @@ NOWDOC_CHARS=({NEWLINE}*(([^a-zA-Z_\x7f-\xff\n\r][^\n\r]*)|({LABEL}[^a-zA-Z0-9_\
     pushState(ST_IN_SCRIPTING);
 }
 
+<ST_IN_SCRIPTING>{ONUM} {
+    // PHP 8.1: Explicit octal integer literal notation
+    // https://wiki.php.net/rfc/explicit_octal_notation
+    return createFullSymbol(ASTPHP5Symbols.T_LNUMBER);
```

Comment:
Just noticed - for `HNUM` and `BNUM` below, we return `T_DNUMBER` - is that correct?

Reply:
It should not be correct. Actually, `T_LNUMBER` or `T_DNUMBER` is returned in PHP. (`ONUM` too) So need more checks. However, I don't think there are big effects on the existing features of NB currently.

Reply:
e.g. 
```php
$hex1 = 0x45fffabcde0000000;
var_dump($hex1); // double(8.0704412748217E+19)

$hex2 = 0x1f;
var_dump($hex2); // int(31)
```

Reply:
OK, thanks.

Reply:
I'll fix it if we come across a problem :)

Reply:
Sure, np.